### PR TITLE
Export on mac.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart [hunk 1]
```diff
@@ -45,13 +49,21 @@ class FileSystemDesktop implements FileIO {
   }
 
   @override
-  void writeStringToFile(
+  void writeStringToFile<T>(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart:52]: rename to writeContentsToFile

File: packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart [hunk 2]
```diff
@@ -45,13 +49,21 @@ class FileSystemDesktop implements FileIO {
...
     bool isMemory = false,
   }) {
-    final logFile = _exportDirectory(isMemory: isMemory).childFile(filename);
-    logFile.writeAsStringSync(contents, flush: true);
+    final file = _exportDirectory(isMemory: isMemory).childFile(filename);
+    if (contents is String) {
+      file.writeAsStringSync(contents, flush: true);
+    } else if (contents is Uint8List) {
+      final path = file.path;
+      final XFile xFile = XFile.fromData(contents, name: path);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart:62]: remove type on LHS

Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart:62]: can we just use `file.writeAsBytesSync` instead?

File: packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart [hunk 3]
```diff
@@ -45,13 +47,19 @@ class FileSystemDesktop implements FileIO {
...
+    T contents, {
     bool isMemory = false,
   }) {
-    final logFile = _exportDirectory(isMemory: isMemory).childFile(filename);
-    logFile.writeAsStringSync(contents, flush: true);
+    final file = _exportDirectory(isMemory: isMemory).childFile(filename);
+    if (contents is String) {
+      file.writeAsStringSync(contents, flush: true);
+    } else if (contents is Uint8List) {
+      file.writeAsBytesSync(contents);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/file/_file_desktop.dart:59]: set flush: true

File: packages/devtools_app/lib/src/shared/config_specific/file/_file_web.dart [hunk 4]
```diff
@@ -21,12 +21,16 @@ class FileSystemWeb implements FileIO {
...
   @override
-  void writeStringToFile(
+  void writeContentsToFile<T>(
     String filename,
-    String contents, {
+    T contents, {
     bool isMemory = false,
   }) {
-    _files.putIfAbsent(filename, () => contents);
+    if (contents is String) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/file/_file_web.dart:29]: should this be `if (contents is String || contents is Uint8List)`?

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/config_specific/file/_file_web.dart:29]: Nope. For web we use different mechanism for Uint8List, merged earlier today.
I am not even sure that we use _file_web.dart even for String. Most likely, no.

**Pull request merged**